packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2948,7 +2958,12 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
     if (_showBodyScrim) {
       _addIfNonNull(
         children,
-        ModalBarrier(dismissible: false, color: _bodyScrimColor),
+        widget.bottomSheetScrimBuilder != null
+            ? widget.bottomSheetScrimBuilder!(context, _bodyScrimAnimation)
+            : ModalBarrier(
+              dismissible: false,
+              color: _bodyScrimColor.withOpacity(_bodyScrimAnimation),
+            ),
```

Comment:
Would it be possible to make bottomSheetScrimBuilder default to a builder that builds this ModalBarrier, and make it non-nullable?